ledger/acctupdates.go

```diff
@@ -387,7 +406,13 @@ func (au *accountUpdates) lookupKv(rnd basics.Round, key string, synchronized bo
 
 		persistedData, err := au.accountsq.lookupKeyValue(key)
 		if persistedData.round == currentDbRound {
-			return persistedData.value, nil
+			if persistedData.value != nil {
+				// if we read actual data return it
+				au.baseBoxes.writePending(persistedData, key)
+				return persistedData.value, err
+			}
+			// otherwise return empty
+			return nil, err
 		}
 
 		// The db round is unexpected...
```

Comment:
@algoidurovic Leaving another note on an unrelated line because impacted areas are not in the diff.

Double checking my understanding:  Is it accurate to expect `au.baseBoxes` updates in `postCommit`?

I see `au.baseResources` updates happening based on return types from `makeCompactResourceDeltas`.

Reply:
Hopefully, this has been addressed in the latest commits. The new test in `accountupdates_test.go` does fail on the code when your comment was written, and now is passing after implementation gaps were filled.

Reply:
@algoidurovic New test in `accountupdates_test.go` covers it from my end - thanks + resolving.